File: python/poo/banco/src/usuario.py
""""Classe Reponsável por Criar um Usuário para o sistema """
class Usuario:
    def __init__(self, nome, CPF, telefone, chave_pix):
        self.__nome = nome
        self.__CPF = CPF
        self.__telefone = telefone
        self.__chave_pix = chave_pix
        self.__saldo = 0

    """" Métodos Setters e Getters, Útilizados Para pegar o definir novos valores, CPF é identificador Único"""
    def set_nome(self, novo_nome):
        self.__nome = novo_nome
    def set_telefone(self, novo_telefone):
        self.__telefone = novo_telefone
    def set_chave_pix(self, nova_chave):
        self.__chave_pix = nova_chave
    def soma_saldo(self, novo_saldo):
        self.__saldo += novo_saldo
    def sub_saldo(self, novo_saldo):
        self.__saldo -= novo_saldo

    def get_nome(self):
        return self.__nome
    def get_CPF(self):
        return self.__CPF
    def get_telefone(self):
        return self.__telefone
    def get_chave_pix(self):
        return self.__chave_pix
    def get_saldo(self):
        return self.__saldo

File: Java/poo/banco/src/Usuario.java
package src;
/**
 * @author José Kauan Pereira
*/
public class Usuario{
	private String nome;
	private final String CPF;
	private double saldo;
	private String pix;
	private String telefone;

	public Usuario(String nome, String CPF, String pix, String telefone){
		this.nome = nome;
		this.CPF = CPF;
		this.pix = pix;
		this.telefone = telefone;
		this.saldo = 0;
	}

	/**
	 * Métodos Setters
	 * */
	public void setNome(String nome){
		this.nome = nome;
	}
	public void somaSaldo(double saldo){
		this.saldo += saldo;
	}
	public void subSaldo(double saldo){
		this.saldo -= saldo;
	}
	public void setPix(String pix){
		this.pix = pix;
	}
	public void setTelefone(String telefone){
		this.telefone = telefone;
	}

	/**
	 * Métodos Getters
	 * */
	 public String getNome(){
	 	return this.nome;
	 }
	 public String getCPF(){
	 	return this.CPF;
	 }
	 public double getSaldo(){
	 	return this.saldo;
	 }
	 public String getPix(){
	 	return this.pix;
	 }
	 public String getTelefone(){
	 	return this.telefone;
	 }

	 @Override
	 public boolean equals(Object obj){
	 	if(this == obj){
	 		return true;
	 	}
	 	if(obj == null || this.getClass() != obj.getClass()){
	 		return false;
	 	}
	 	Usuario usu = (Usuario) obj;
	 	return  this.getCPF().equals(usu.getCPF());
	 }

	 @Override
	 public int hashCode(){
	 	return this.getCPF().hashCode();
	 }

	 @Override
	 public String toString(){
	 	return "Nome: " + nome + " \nCPF: " + CPF + "\ntelefone: " + telefone + "\nChave Pix: " + pix + "\nsaldo: " + saldo +" R$\n";
	 }
}	
